icu4c/source/common/lstmbe.cpp

```diff
@@ -809,7 +809,15 @@ U_CAPI const LSTMData* U_EXPORT2 CreateLSTMDataForScript(UScriptCode script, UEr
 
 U_CAPI const LSTMData* U_EXPORT2 CreateLSTMData(UResourceBundle* rb, UErrorCode& status)
 {
-    return new LSTMData(rb, status);
+    if (U_FAILURE(status)) {
+        return nullptr;
+    }
+    const LSTMData* result = new LSTMData(rb, status);
```

Comment:
Need to check for a null pointer return from the new, and set status U_MEMORY_ALLOCATION_ERROR if it happens. The LocalPointer constructor with an error code is convenient for this.